src/puppetlabs/trapperkeeper/services/webserver/jetty9_core.clj

```diff
@@ -952,6 +956,25 @@
                                             [server-id (initialize-context)]))
                  :default-server (get-default-server-from-config config)))
 
+(schema/defn ^:always-validate reload-crl-on-change!
+  "Reload the CRL file used by the supplied ssl-context-factory whenever any
+  changes to the file occur."
+  [ssl-context-factory :- InternalSslContextFactory
+   watcher :- (schema/protocol watch-protocol/Watcher)]
+  (when-let [crl-path (.getCrlPath ssl-context-factory)]
+    (let [normalized-crl-path (.getCanonicalPath (fs/file crl-path))]
+      (watch-protocol/add-watch-dir! watcher (fs/parent crl-path)
```

Comment:
For reference, I submitted a ticket for the work to add support for non-recursive directory watching to the tk-filesystem-watcher service, [TK-448](https://tickets.puppetlabs.com/browse/TK-448).  It might be reasonable to consider merging this ahead of that work getting done.